src/main/java/com/redhat/devtools/lsp4ij/features/workspaceSymbol/LSPWorkspaceTypeDeclarationProvider.java

```diff
@@ -0,0 +1,118 @@
+
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.isDoneNormally;
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.waitUntilDone;
+
+/**
+ * Implements the IDE's standard Go To Type Declaration action using LSP textDocument/typeDefinition. -->
+ */
+public class LSPWorkspaceTypeDeclarationProvider implements TypeDeclarationPlaceAwareProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LSPWorkspaceTypeDeclarationProvider.class);
+
+    @Override
+    public PsiElement @Nullable [] getSymbolTypeDeclarations(@NotNull PsiElement symbol) {
+        // Not much we can do without an offset
+        return PsiElement.EMPTY_ARRAY;
+    }
+
+    @Override
+    public PsiElement @Nullable [] getSymbolTypeDeclarations(@NotNull PsiElement symbol, Editor editor, int offset) {
+        if (!symbol.isValid()) {
+            return PsiElement.EMPTY_ARRAY;
+        }
+
+        Project project = symbol.getProject();
+        if (project.isDisposed()) {
+            return PsiElement.EMPTY_ARRAY;
+        }
+
+        PsiFile file = symbol.getContainingFile();
+        if ((file == null) || !file.isValid()) {
+            return PsiElement.EMPTY_ARRAY;
+        }
+
+        if (ProjectIndexingManager.canExecuteLSPFeature(file) != ExecuteLSPFeatureStatus.NOW) {
+            return PsiElement.EMPTY_ARRAY;
+        }
+
+        if (!LanguageServiceAccessor.getInstance(project)
+                .hasAny(file.getVirtualFile(), ls -> ls.getClientFeatures().getTypeDefinitionFeature().isTypeDefinitionSupported(file))) {
+            return PsiElement.EMPTY_ARRAY;
+        }
+
+        Document document = PsiDocumentManager.getInstance(project).getDocument(file);
+        if (document == null) {
+            return PsiElement.EMPTY_ARRAY;
+        }
+
+        LSPTypeDefinitionSupport typeDefinitionSupport = LSPFileSupport.getSupport(file).getTypeDefinitionSupport();
+        var params = new LSPTypeDefinitionParams(LSPIJUtils.toTextDocumentIdentifier(file.getVirtualFile()), LSPIJUtils.toPosition(offset, document), offset);
+        CompletableFuture<List<Location>> typeDefinitionsFuture = typeDefinitionSupport.getTypeDefinitions(params);
+        try {
+            waitUntilDone(typeDefinitionsFuture, file);
+        } catch (ProcessCanceledException ex) {
+            // cancel the LSP requests textDocument/typeDefinition
+            typeDefinitionSupport.cancel();
+        } catch (CancellationException ex) {
+            // cancel the LSP requests textDocument/typeDefinition
+            typeDefinitionSupport.cancel();
+        } catch (ExecutionException e) {
+            LOGGER.error("Error while consuming LSP 'textDocument/typeDefinition' request", e);
+        }
+
+        if (isDoneNormally(typeDefinitionsFuture)) {
+            List<Location> typeDefinitions = typeDefinitionsFuture.getNow(null);
+            if (!ContainerUtil.isEmpty(typeDefinitions)) {
```

Comment:
I think we should show an hint when there are no type definition (like you did for implementation)